pkg/runtime/wfengine/workflow.go

```diff
@@ -357,10 +363,15 @@ func (wf *workflowActor) runWorkflow(ctx context.Context, actorID string, remind
 
 			resp, err := wf.actors.Call(ctx, req)
 			if err != nil {
-				return newRecoverableError(
-					fmt.Errorf("failed to invoke activity actor '%s' to execute '%s': %v", targetActorID, ts.Name, err))
+				if errors.Is(err, ErrDuplicateInvocation) {
+					wfLogger.Warnf("%s: activity invocation %s#%d was flagged as a duplicate and will be skipped", actorID, ts.Name, e.EventId)
```

Comment:
Suggest adding a `continue` after this call, for 3 reasons:

1. Safer in case someone decided to add more code in the loop in the future, after this block
1. Then you don't need the `if resp != nil` check that was added
1. It makes the flow more clear

Reply:
I considered that but then I have to worry about the `defer resp.Close()` call never getting executed, right? This seemed like a simpler approach that won't make the linter that tracks response-closing angry.

Reply:
If err != nil, resp should be nil (or at least shouldn't need closing) per Go convention.

Reply:
Also since you mention that, worth pointing out that because this line is in a loop, the defer will be executed at the end of the method. Not sure if that could be an issue and maybe an immediate `resp.Close()` (without defer) may be better

Reply:
Okay, I went ahead and switched to using `continue`. I might have been overthinking this. I do like that the logic is much simpler now.